packages/myst-cli/src/build/pdf/create.ts

```diff
@@ -178,11 +149,153 @@ export async function createPdfGivenTexExport(
       await copyFile(texLogBuild, texLogOutput);
     }
   }
+
   const logFiles = copyLogs ? [logOutput, texLogOutput] : [logBuild, texLogBuild];
   if (!fs.existsSync(pdfOutput)) {
     const err = Error(`Error exporting: ${pdfOutput}`);
     (err as any).logFiles = logFiles;
     throw err;
   }
+  
   return { logFiles, tempFolders: [buildPath] };
 }
+
+function ensurePaths(session: ISession, texOutput: string, pdfOutput: string) {
+  const buildPath = createTempFolder(session);
+  const texFile = path.basename(texOutput);
+  const texBuild = path.join(buildPath, texFile);
+  copyContents(path.dirname(texOutput), buildPath);
+
+  if (!fs.existsSync(texBuild)) {
+    throw Error(`Error exporting: ${pdfOutput}\nCould not find tex file: ${texOutput}`);
+  }
+
+  const pdfBasename = path.basename(pdfOutput, path.extname(pdfOutput));
+  const pdfFile = `${pdfBasename}.pdf`;
+  const pdfBuild = path.join(buildPath, pdfFile);
+
+  const logFile = `${pdfBasename}.log`;
+  const texLogFile = `${pdfBasename}.shell.log`;
+  // Temporary log file locations
+  const logBuild = path.join(buildPath, logFile);
+  const texLogBuild = path.join(buildPath, texLogFile);
+  // Log file location saved alongside pdf
+  const logOutputFolder = getLogOutputFolder(pdfOutput);
+  const logOutput = path.join(logOutputFolder, logFile);
+  const texLogOutput = path.join(logOutputFolder, texLogFile);
+
+  return {
+    buildPath,
+    texFile,
+    texLogFile,
+    pdfBuild,
+    logBuild,
+    texLogBuild,
+    logOutputFolder,
+    logOutput,
+    texLogOutput,
+  };
+}
+
+async function runCommands(
+  session: ISession,
+  template: string | null | undefined,
+  texFile: string,
+  texLogFile: string,
+  templateLogString: string,
+  pdfBuild: string,
+  buildPath: string,
+  glossaries?: boolean,
+) {
+  const toc = tic();
+  let buildError = await runPdfBuildCommand(session, texFile, texLogFile, templateLogString, template, pdfBuild, buildPath);
+
+  if (buildError || !glossaries) {
+    return { buildError, toc };
+  }
+
+  // Glossaries require two more commands to be run
+  buildError = await runGlossariesBuildCommand(session, texFile, buildPath);
```

Comment:
🤔 Now that there are multiple commands that need to be run in a specific order, I'm wondering if some of this logic should move to jtex... Not great, though, because we probably don't want all the CLI logging stuff, etc in jtex?

I suppose for now, leaving all this here is fine; if more build commands get into the mix with subsequent features, we will probably need to refactor a bit.